client/src/app/components/forms/rsvp/rsvp.component.ts

```diff
@@ -21,7 +26,18 @@ export class RsvpFormComponent {
 		guests: this.fb.array([])
 	});
 
-	constructor(private fb: NonNullableFormBuilder){}
+	constructor(private fb: NonNullableFormBuilder, private route: ActivatedRoute, private api: RsvpService){}
+
+	ngOnInit() {
+		const codeInput = this.route.snapshot.queryParamMap.get('code');
+		if(codeInput) {
+			this.code = codeInput;
+			this.api.getRSVPFromCode(this.code).subscribe((data: any) => {
```

Comment:
I'd type this data coming back - this should be a RecursivePartial<guestData>, right?

```suggestion
			this.api.getRSVPFromCode(this.code).subscribe((data: RecursivePartial<guestData>) => {
```

Reply:
oh yeah lol